task-requests/script.js

```diff
@@ -109,6 +122,60 @@ const changeFilter = () => {
   taskRequestContainer.innerHTML = '';
 };
 
+function handleBadgeDeletion(badgeType, badgeContent) {
+  let newFilterType;
+  if (Array.isArray(filterStates[badgeType])) {
+    newFilterType = filterStates[badgeType].filter(
+      (type) => type !== badgeContent,
+    );
+  } else if (filterStates[badgeType] === badgeContent) {
+    newFilterType = '';
+  }
+  updateFilterStates(badgeType, newFilterType);
+  const checkboxValue =
+    badgeType === 'requestType' ? badgeContent : badgeContent.toUpperCase();
+  uncheckCheckbox(checkboxValue);
+}
+
+function deleteBadge(e, badgeType) {
+  const badgeContent = e.target.textContent;
+  handleBadgeDeletion(badgeType, badgeContent);
+}
+
+function showBadges() {
+  badgesContainer.innerHTML = '';
+  const statuses = filterStates?.status;
+  const requestTypes = filterStates?.requestType;
+
+  if (statuses) {
+    if (Array.isArray(statuses)) {
+      statuses.forEach((status) => {
+        const badge = createBadge(status, 'status');
+        badgesContainer.appendChild(badge);
+      });
+    } else if (statuses) {
+      const badge = createBadge(statuses, 'status');
+      badgesContainer.appendChild(badge);
+    }
```

Comment:
`showBadges` function could be optimized by using a single loop to iterate over both status and request type filters instead of separate loops for each.

Reply:
done, optimized it to use single loop for iterating over both filters.